src/Features/Core/Portable/ExtractMethod/SelectionResult.cs

```diff
@@ -138,6 +139,61 @@ public bool ShouldPutAsyncModifier()
             return false;
         }
 
+        public bool ShouldCallConfigureAwaitFalse()
+        {
+            var firstToken = GetFirstTokenInSelection();
+            var lastToken = GetLastTokenInSelection();
+
+            var span = TextSpan.FromBounds(firstToken.SpanStart, lastToken.Span.End);
+
+            foreach (var node in SemanticDocument.Root.DescendantNodesAndSelf())
+            {
+                if (!node.Span.OverlapsWith(span))
+                {
+                    continue;
+                }
+
+                if (IsConfigureAwaitFalse(node) && !UnderAnonymousOrLocalMethod(node.GetFirstToken(), firstToken, lastToken))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private bool IsConfigureAwaitFalse(SyntaxNode node)
+        {
+            var syntaxFacts = SemanticDocument.Project.LanguageServices.GetService<ISyntaxFactsService>();
+            if (!syntaxFacts.IsInvocationExpression(node))
+            {
+                return false;
+            }
+
+            var invokedExpression = syntaxFacts.GetExpressionOfInvocationExpression(node);
+            if (!syntaxFacts.IsSimpleMemberAccessExpression(invokedExpression))
+            {
+                return false;
+            }
+
+            var name = syntaxFacts.GetNameOfMemberAccessExpression(invokedExpression);
+            var identifier = syntaxFacts.GetIdentifierOfSimpleName(name);
+            if (!syntaxFacts.StringComparer.Equals(identifier.ValueText, nameof(Task.ConfigureAwait)))
```

Comment:
💡 This should work for `ValueTask` as well, but it would be good to add a test for this case (where the original code awaits the result of a `ValueTask` and uses `ConfigureAwait(false)`).